src/main/java/com/stories/service/StoriesServiceImpl.java

```diff
@@ -62,13 +60,28 @@ public void deleteStory(String id) throws Exception {
 
 	public StoryDomain updateStory(StoryDomain storyDomain, String id) throws Exception {
 		StoryModel storyModel = mapperFacade.map(storyDomain, StoryModel.class);
-		if (!storiesRepository.existsById(id))
-			throw new EntityNotFoundException("Story not found", StoryDomain.class);
-		storyModel.set_id(id);
-		storiesRepository.save(storyModel);
-		storyDomain = mapperFacade.map(storyModel, StoryDomain.class);
-		logger.debug("Updating story with the id: " + id + " - JSON : {}", storyDomain);
-		return storyDomain;
+		SprintsClient sprintClient = new SprintsClient();
+		boolean sprintExists = sprintClient.existsSprintById(storyDomain.getSprint_id());
+		if (true == sprintExists) {
```

Comment:
If sprintExists return a Boolean value, you don´t need validate again with a true value in the "IF"

Your IF should be like =   if(sprintExists)